src/compiler/tsbuild.ts

```diff
@@ -944,16 +990,40 @@ namespace ts {
                 return;
             }
 
+            if (status.type === UpToDateStatusType.UpToDateWithUpstreamTypes) {

```

Comment:
how is this new status related to watch mode? It sounds unrelated.

Reply:
If you have `proj` depending on `dep` when change happens to the `dep` thats local (not changing its d.ts file) the file will get updated stamp and hence the status of this build would be `UpToDateWithUpstreamTypes` because of our own knowledge that the change was local and .d.ts not really changed. But if we did not update the type stamps, if we kill build at this point and new process starts, it wont have that info and it would create new program and emit etc. So this was existing but that got fixed as part of making sure dependent projects have correct time stamps in case of incremental build (Note how we have to update timestamps of unchanged emit files when doing incremental emit to ensure it doesnt falsely assumed to be not uptodate when new process with tsc -b starts where this info about the d.ts updates etc is not in the memory)

Reply:
So this is more about making sure that incrementally built and done projects dont unnecessarily start creating and emitting when tsc -b is invoked in another process